app/scripts/models/auth_brokers/fx-sync-channel.js

```diff
@@ -0,0 +1,257 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A broker that communicates with Firefox.
+ * Concrete sub-classes should define the function `createChannel`
+ * and a `commands` object.
+ */
+
+define(function (require, exports, module) {
+  'use strict';
+
+  const _ = require('underscore');
+  const AuthErrors = require('lib/auth-errors');
+  const ChannelMixin = require('models/auth_brokers/mixins/channel');
+  const Cocktail = require('cocktail');
+  const FxSyncAuthenticationBroker = require('models/auth_brokers/fx-sync');
+  const Logger = require('lib/logger');
+  const p = require('lib/promise');
+
+  const proto = FxSyncAuthenticationBroker.prototype;
+
+  const FxSyncChannelAuthenticationBroker = FxSyncAuthenticationBroker.extend({
+    type: 'fx-sync-channel',
+
+    /**
+     * Must be overridden with an object that contains:
+     *
+     * {
+     *   CAN_LINK_ACCOUNT: <specify in subclass>,
+     *   CHANGE_PASSWORD: <specify in subclass>,
+     *   DELETE_ACCOUNT: <specify in subclass>,
+     *   LOADED: <specify in subclass>,
+     *   LOGIN: <specify in subclass>
+     * }
+     *
+     * @property commands
+     */
+    commands: null,
+
+    defaultCapabilities: _.extend({}, proto.defaultCapabilities, {
+      sendChangePasswordNotice: true
+    }),
+
+    getCommand (commandName) {
+      if (! this.commands) {
+        throw new Error('this.commands must be specified');
+      }
+
+      var command = this.commands[commandName];
```

Comment:
These were copied verbatim from the previous fx-sync, though it's a good time to update.